### DISCUSSION TITLE
Allow having a default lock_timeout in migrations

### DISCUSSION THREAD

**Discussion by czynskee** [jeremyevans/sequel#2160] [General]
In Postgres (and maybe other DBMSs - I'm not sure), it's possible for a simple migration to create big problems due to lock pileups. A way to prevent this from causing production incidents is to use `lock_timeout`. [This article](https://postgres.ai/blog/20210923-zero-downtime-postgres-schema-migrations-lock-timeout-and-retries) describes the problem and solution in detail. 

It's simple enough to add this command to individual migrations. It would be nice if we could have it added to every migration by default though, so we don't have to remember to add it. At the same time, we would like to have a way to override it in certain cases - either within a certain migration or via an option to `Migrator.run`.

Basically, I'm looking for the equivalent of [this](https://github.com/procore/migration-lock-timeout) plugin for ActiveRecord. 

I found [this](https://github.com/jeremyevans/sequel/discussions/2092) discussion, but I believe that will only work for setting the `statement_timeout`. The `lock_timeout` can only be set after the transaction is opened.

**Comment by jeremyevans** [comment]
Thanks for bringing this up. This seems like a good feature to add to Sequel.  I've added it to my to-do list. If someone wants to work on it, that may speed things up.
